core/kernel/pathname_spec.rb

```diff
@@ -0,0 +1,26 @@
+require_relative '../../spec_helper'
+require 'pathname'
+
+describe "Kernel#Pathname" do
+  it "is a public method" do
+    Kernel.public_methods.should include(:Pathname)
+  end
+
+  ruby_version_is ''...'2.7' do
+    it "returns a new pathname when called with a pathname argument" do
+      path = Pathname('foo')
+      new_path = Pathname(path)
+
+      (path.object_id == new_path.object_id).should be_false
+    end
+  end
+
+  ruby_version_is '2.7' do
+    it "returns same argument when called with a pathname argument" do
+      path = Pathname('foo')
+      new_path = Pathname(path)
+
+      (path.object_id == new_path.object_id).should be_true
+    end
```

Comment:
Looks good but it's common to use `equal` matcher to check whether objects are the same object:

- https://github.com/ruby/mspec/blob/master/lib/mspec/matchers/equal.rb#L8
- https://ruby-doc.org/core-2.7.2/BasicObject.html#method-i-equal-3F

Reply:
done! thanks